conda-env/prod.yml

```diff
@@ -0,0 +1,37 @@
+# Conda production environment for datasm operations.
+name: datasm_prod
+channels:
+  - conda-forge
+  - e3sm
+  - defaults
+dependencies:
+  # Base
+  # ==================
+  - python >=3.9
+  - pip
+  - distributed
+  - ipdb
+  - matplotlib
+  - netcdf4
+  - numpy >=1.23.0 # This version of numpy includes support for Python 3.11.
+  - pyyaml
+  - termcolor
+  - tqdm
+  - watchdog
+  - xarray >=2022.02.0 # This version of Xarray drops support for Python 3.8.
+  # Required for CWL workflows.
+  - cwltool >=3.1.20220202173120
+  - nodejs >=17.4.0
+  # Used in modules for `extract`, `validate`, and `postprocess` operations.
+  - nco >=5.1.3
+  - e3sm_to_cmip >=1.9.1
+  - zstash >=1.2.0
+  # Testing
+  # ==================
+  - pytest >=7.1.1
+  # Quality Assurance
+  # ==================
+  - black >=22.3.0
+  - pip:
+    - esgcet>=5.2.0
```

Comment:
`esgcet>=5.2.0` now includes an `-xarray` flag to replace `autocurator`. I removed `autocurator` as a result.

Reply:
Wow - Thanks Tom!  This is great.  I have some 5/6 different environments (some for dev, some for pub), some customized for v1 or v2 Large Ensemble processing, some with/without e2c v1.10.0rc1 (nee rc2) - and it will be great to cut them down.

I have long-running CMIP6 jobs running (expected to complete around Sept 5) and just finished a publication run) so I'll need to create a new environment (or two) to test these - don't want to destabilize running stuff...

I need to think about how to test this (publish without actually publishing, etc) or else publish something small.

Reply:
Given that there as never been a datasm "stable release", this would leave me with only a "dev" environment ...

The scope of datasm is so broad, no regression testing makes headway before operational exigencies demand changes to accommodate new data irregularities.  Not to mention - I have scores of big jobs queued up that are pushing against slurm "PENDING (resources)".  I suppose I need to develop a "non-slurm" test regime.

Reply:
@tomvothecoder Although the prod.yml includes

`- cwltool >=3.1.20220202173120`

when I build a prod environment and pip install datasm (and e3sm_to_cmip, FWIW), datasm postprocess fails with 

`/var/spool/slurmd/job210469/slurm_script: line 7: cwltool: command not found`

The environment I built shows (mamba list):

```
cryptography              41.0.3          py310h75e40e8_0    conda-forge
curl                      8.2.1                hca28451_0    conda-forge
cwl-upgrader              1.2.8              pyhd8ed1ab_0    conda-forge
cwl-utils                 0.28               pyh1d7be83_0    conda-forge
cwlformat                 2022.02.18         pyhd8ed1ab_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
```
(not that I ever looked for cwltool before, so I don't know what to expect.)

Reply:
@tomvothecoder This could be my fault (naturally...).  I changed "- default" to "- nodefault", having read a comment about that.  Now that I've changed it back and rebuilt, cwltool appears... magic!

Reply:
Could be superstition on my part.  Other intervening changes may have occurred.  I should create another env with "nodefaults" and check again.  Maybe the (acme1) base environment is different.  (to many variables...)